crates/task-impls/src/consensus.rs

```diff
@@ -1062,12 +1071,18 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                     ))
                     .await;
 
-                // Add to the storage that we have received the VID disperse for a specific view
-                self.consensus
-                    .write()
-                    .await
-                    .vid_shares
-                    .insert(view, disperse.clone());
+                {
+                    let mut consensus_lock = self.consensus.write().await;
+                    // Add to the storage that we have received the VID disperse for a specific view
+                    if let Some(vid_disperse) = consensus_lock.vid_shares.get_mut(&view) {
+                        vid_disperse.insert(disperse.data.recipient_key.clone(), disperse.clone());
+                    } else {
+                        let mut proposals_map = HashMap::new();
+                        proposals_map.insert(disperse.data.recipient_key.clone(), disperse.clone());
+                        consensus_lock.vid_shares.insert(view, proposals_map);
+                    }
```

Comment:
nit: Can be simplified with `.entry.or_default` instead of get_mut.

https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html#method.or_default